modules/node/keeper/peer_filter.go

```diff
@@ -0,0 +1,30 @@
+package keeper
+
+import (
+	"encoding/hex"
+
+	abci "github.com/tendermint/tendermint/abci/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/bianjieai/iritamod/modules/node/types"
+)
+
+// FilterNodeByID implements sdk.PeerFilter
+func (k Keeper) FilterNodeByID(ctx sdk.Context, nodeID string) abci.ResponseQuery {
+	id, err := hex.DecodeString(nodeID)
+	if err != nil {
+		return abci.ResponseQuery{
+			Code: types.ErrInvalidNodeID.ABCICode(),
+		}
+	}
+
+	_, found := k.GetNode(ctx, id)
+	if !found {
+		return abci.ResponseQuery{
+			Code: types.ErrUnknownNode.ABCICode(),
+		}
+	}
+
```

Comment:
We should add a whitelist enable param,
or if no nodes are set, return `abci.ResponseQuery{}`.

Reply:
do it via another PR